src/app/components/home/product-results/product-results.component.ts

```diff
@@ -10,5 +10,5 @@ import ProductInterface from '../../../interfaces/products';
   styleUrl: './product-results.component.scss',
 })
 export class ProductResultsComponent {
-  @Input() products: ProductInterface[] = [];
+  @Input() products: ProductInterface[] | null | undefined = [];
```

Comment:
I'm not 100% sure it's necessary for this at all. In the parent you can have your async pipe and then fall back to empty.
i.e.
`[products]="(products$ | async) ?? []"`

Reply:
Or as Hugo said, wrap in an `@if` block